update.ps1

```diff
@@ -343,26 +426,38 @@ try {
             'UpdateEmployeeAccount' {
                 Write-Information "Employee Account property(s) required to update: $($propertiesEmployeeChanged.Name -join ', ')"
                 $updateQueryProperties = @()
-                switch ($propertiesEmployeeChanged.name) {
-                    'BirthDate' { $updateQueryProperties += "BirthDate = #$($actionContext.Data.Employee.BirthDate)#"; continue }
-                    'Language' { $updateQueryProperties += "Language = $($actionContext.Data.Employee.Language)"; continue }
-                    'SalaryNR' { $updateQueryProperties += "SalaryNR = '$($actionContext.Data.Employee.SalaryNR)'"; continue }
-                    'HireDate' { $updateQueryProperties += "HireDate = #$($actionContext.Data.Employee.HireDate)#" ; continue }
-                    'TerminationDate' { $updateQueryProperties += "TerminationDate = #$($actionContext.Data.Employee.TerminationDate)#"; continue }
+                foreach ($prop in $propertiesEmployeeChanged) {
+                    $updateQueryProperties += switch ($prop.Name ) {
+                        'BIRTHDATE' { "BirthDate = $( if (-not ([string]::IsNullOrEmpty($prop.value))) { "#$($prop.value)#" } else{ 'NULL' } )" ; break }
+                        'LANGUAGE' { "Language = $($prop.value)" ; break }
+                        'SALARYNR' { "SalaryNR = '$($prop.value)'" ; break }
+                        'HIREDATE' { "HireDate = $( if (-not ([string]::IsNullOrEmpty($prop.value))) { "#$($prop.value)#" } else{ 'NULL' } )" ; break }
+                        'TERMINATIONDATE' { "TerminationDate = $( if (-not ([string]::IsNullOrEmpty($prop.value))) { "#$($prop.value)#" } else{ 'NULL' } )" ; break }
+                    }
                 }
+
                 $updateQueryEmployee = "
                     UPDATE
                         EMPLOYEE
                     SET
                         $($updateQueryProperties -join ",`n" )
                     Where
-                        EmployeeId = '$($employeeObject.employeeId)"
+                        EmployeeId = $($employeeObject.employeeId)"
 
                 # Make sure to test with special characters and if needed; add utf8 encoding.
                 if (-not($actionContext.DryRun -eq $true)) {
-                    Write-Information "Updating  account with accountReference: [$($actionContext.References.Account)]"
-                    $updatedEmployeeResult = Invoke-IProtectQuery -JSessionID $jSessionID -WebSession $webSession -Query $updateQueryEmployee -QueryType 'update'
-                } else {
+                    Write-Information "Updating IProtect Employee account with accountReference: [$($actionContext.References.Account)]"
```

Comment:
Deze auditlog wordt alleen getoond na een update trigger met een correlatie, wil je deze melding dan inderdaad niet zien?
En wordt er dan niet alsnog een standaard auditlog getoond in HelloId?

Verder wordt de auditlog onderdrukt zolang Data and previousData gelijk zijn.

$outputContext.AuditLogs.Add([PSCustomObject]@{
                Message = 'No changes will be made to the account during enforcement'
                IsError = $false
            })